### PR TITLE
Note_Actions

### PR DESCRIPTION
**Pull request by raieegulhane**
- add note
- archive note
- restore from archive
- trash note
- restrore from trash
- permanently delete note
- add label to notes
- delete label from note

### CODE REVIEW

File: src/components/notes/note-input-form/index.jsx [hunk 1]
```diff
@@ -4,16 +4,18 @@ import { RichTextEditor } from "../text-editor";
 import { ColorPalette } from "../color-palette";
 import { PriorityList } from "../priority-list";
 import { LabelEditor } from "../../label-editor";
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import { postNoteService } from "../../../services/notes-services/post-note-service";
 import { useAuth } from "../../../contexts/auth-context";
```
Comment (veereshbv04) [inline src/components/notes/note-input-form/index.jsx]: Suggestion: create an index page and import all contexts here. and through this you can import any contextx in a single line like
import { useAuth, useNote } from "../../../contexts";

File: src/components/notes/note-card/note-card.css [hunk 2]
```diff
@@ -1,12 +1,17 @@
 .note-card-wrapper {
     position: relative;
-    border: 1px solid var(--secondary-grey);
+    color: var(--text-color);
+    width: 250px;
+    height: 200px;
```
Comment (ShrutiLakhani) [inline src/components/notes/note-card/note-card.css:5]: Please use rem instead of px

File: src/components/notes/note-input-form/index.jsx [hunk 3]
```diff
@@ -1,74 +1,210 @@
...
+    const { noteState, noteDispatch } = useNote();
+    const { allNotes, isEditing, isEditingId } = noteState;
+
+    const { componentState, componentDispatch } = useComponent();
+    const { showColorPalette, showPriorityOptions, showLabelEditor } = componentState;
+
+    const initialNoteValues = {
+        noteTitle: "",
+        noteBody: "",
+        noteColor: "#ffffff",
```
Comment (ShrutiLakhani) [inline src/components/notes/note-input-form/index.jsx:20]: You may want to use color variables

File: src/components/notes/note-input-form/note-input-form.css [hunk 4]
```diff
@@ -5,19 +5,30 @@
...
     left: 0;
+    overflow-y: auto;
+    z-index: 5;
 }
 
 .note-input-container {
     position: relative;
     background-color: var(--white);
     height: fit-content;
     max-width: 750px;
```
Comment (ShrutiLakhani) [inline src/components/notes/note-input-form/note-input-form.css:16]: use rem

**Review by ShrutiLakhani — APPROVED** [review]
Looks good to merge. Code is clean and readable. Good Job 👍 
Made few suggestions, you may want to have a look

**Review by saicharan-gouru — APPROVED** [review]
-UI looks so cool.
-You can add a screenshot or gif for reference while raising PR.
-Good use of favicon.
-Functionality works fine.

**Pull request merged**